packages/x-components/src/x-modules/experience-controls/store/actions/fetch-controls.action.ts

```diff
@@ -1,37 +1,20 @@
 import { ExperienceControlsXStoreModule } from '../types';
+import { XPlugin } from '../../../../plugins/x-plugin';
 
 /**
  * Default implementation for the {@link ExperienceControlsActions.fetchControls}.
  *
  * @param _context - The {@link https://vuex.vuejs.org/guide/actions.html | context} of the actions,
  * provided by Vuex.
- * TODO: update when adapter is updateds.
- * @param request
- * @returns
+ *
+ * @param request - Parameters are empty for this action.
+ * @returns 'controls' data with the controls data configuration.
  *
  * @public
  */
-
 // eslint-disable-next-line max-len
 export const fetchExperienceControlsResponse: ExperienceControlsXStoreModule['actions']['fetchExperienceControlsResponse'] =
-  async () => {
-    const response = await fetch(
-      // eslint-disable-next-line max-len
-      'https://config-service.internal.test.empathy.co/public/configs?service=xcontrols&instance=empathy'
-    );
-
-    if (response.ok) {
-      const controls = await response.json();
-
-      const aux = {
-        controls,
-        events: {}
-      };
-
-      return aux;
-    } else {
-      throw new Error('Failed to fetch data');
-    }
-
-    //return XPlugin.adapter.experienceControls(request).then(({ controls }) => controls);
+  async (_context, request) => {
+    const { controls } = await XPlugin.adapter.experienceControls(request);
+    return controls;
```

Comment:
This has to return the whole response.
Also, awaiting in an action defeats part of the purpose of the action, when it's done is for very specific reasons.

```suggestion
  (_context, request) => {
    return request
      ? XPlugin.adapter.experienceControls(request).then(response => response)
      : ({} as ExperienceControlsResponse);
```